### PR TITLE
removing result var from context

### PR DESCRIPTION
**Pull request by CodeHarsh**
Fixing https://github.com/rsyslog/rsyslog/issues/3025

### CODE REVIEW

File: contrib/fmhash/fmhash.c [hunk 1]
```diff
@@ -197,10 +195,9 @@ hash_wrapper3(struct svar *__restrict__ const sourceVal, struct svar *__restrict
 		FINALIZE;
 	}
 
-	CHKiRet((hcontext->hash_wrapper_1_2(sourceVal, seedVal, hcontext)));
-	xhash = hcontext->xhash % mod;
-	DBGPRINTF("fmhash: hashXXmod generated hash-mod %" PRIu64 ".", xhash);
-	hcontext->xhash = xhash;
+	CHKiRet((hcontext->hash_wrapper_1_2(sourceVal, seedVal, hcontext, xhash)));
+	(*xhash) = (*xhash) % mod;
```
Comment (rgerhards) [inline contrib/fmhash/fmhash.c]: cppcheck (via Codacy) has a false positive here, it does not seem to understand that the check in 191 protects again `mod==0`. Nevertheless, we should try to work around this issue. How about using `mod = 1` after line 193?

File: contrib/fmhash/fmhash.c [hunk 2]
```diff
@@ -197,10 +195,9 @@ hash_wrapper3(struct svar *__restrict__ const sourceVal, struct svar *__restrict
 		FINALIZE;
```
Comment (rgerhards) [inline contrib/fmhash/fmhash.c]: These two lines could easier (and more up to the code style) be written as
```
ABORT_FINALIZE(RS_RET_ERR);
```
but probably `RS_RET_PARAM_ERROR` would be more up to the point?

File: contrib/fmhash/fmhash.c (hunk 1 above)
Comment (CodeHarsh) [reply contrib/fmhash/fmhash.c]: Adding mod =1 made clang analyzer angry. Adding if(mod!=0) did the trick.

Comment (rgerhards) [reply contrib/fmhash/fmhash.c]: > Adding mod =1 made clang analyzer angry. Adding if(mod!=0) did the trick.

oh, the joys of static analyzers... ;-) Thanks for taking care of this. Will merge most probably tomorrow (now  I really need to run out of office ;-)

**Pull request merged**